### PR TITLE
implemented CSS animations

### PR DESCRIPTION
**Pull request by tzraikov**
#1608

### CODE REVIEW

File: apps/tests/ui/animation/css-animation-tests.ts [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        "animation-duration: 4s; " +
+        "animation-timing-function: ease-in; " +
+        "animation-delay: 1.5; " +
+        "animation-iteration-count: 10; " +
+        "animation-direction: reverse; " +
+        "animation-fill-mode: forwards; " +
+        " }";
+    let animation = createAnimationFromCSS(css);
+
+    TKUnit.assert(animation.name === "first");
```
Comment (hshristov) [inline apps/tests/ui/animation/css-animation-tests.ts]: assertEqual produce better error message

File: apps/tests/ui/view/view-tests-common.ts [hunk 2]
```diff
@@ -201,6 +201,7 @@ export var test_addView_WillThrow_IfView_IsAlreadyAdded = function () {
         }
 
         TKUnit.assert(thrown);
+        views[1]._removeView(newButton);
```
Comment (hshristov) [inline apps/tests/ui/view/view-tests-common.ts]: If this test throws - view won't be added to parent, why removing it here?

File: ui/animation/keyframe-animation.d.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
+declare module "ui/animation/keyframe-animation" {
+
+import view = require("ui/core/view");
+
+    export class KeyframeDeclaration {
```
Comment (hshristov) [inline ui/animation/keyframe-animation.d.ts]: Do we need class here or interface will work too?
Same for KeyframeInfo.

File: ui/animation/keyframe-animation.ts [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            }
+        }
+        for (let index = 1; index < length; index++) {
+            let a = animations[index];
+            if (a["curve"] === undefined) {
+                a["curve"] = info.curve;
+            }
+        }
+        if (animations.length > 0) {
+            animations[0]["duration"] = 0.001;
```
Comment (hshristov) [inline ui/animation/keyframe-animation.ts]: Do we need this duration here?

File: ui/animation/keyframe-animation.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    private animate(view: view.View, index: number, iterations: number) {
+        if (index === 0) {
+            let animation = this.animations[0];
+            let modifier = animation["valueSource"];
+
+            if (animation["backgroundColor"] !== undefined) {
+                view.style._setValue(style.backgroundColorProperty, animation["backgroundColor"], modifier);
+            }
+            if (animation["scale"] !== undefined) {
```
Comment (hshristov) [inline ui/animation/keyframe-animation.ts]: Do we always have to set both 'x' and 'y'?
What will happen if I set only 'x'? Will this reset my 'y' value?

**Comment by hshristov** [comment]
:+1:

**Comment by PanayotCankov** [comment]
:+1:

**Comment by tzraikov** [comment]
run ci

**Pull request merged**